### PR TITLE
SDIT-1751: 🔧 Tweak connection pool settings to reduce invalid connections

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/resources/application-nomis.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ spring:
     datasource:
       hikari:
         maximum-pool-size: 10
-        connectionTimeout: 5000
+        connectionTimeout: 2500
```
Comment (petergphillips) [inline src/main/resources/application-nomis.yml:22]: this was originally 2000, changing to 2500 to match the primary

File: helm_deploy/prison-api/values.yaml [hunk 2]
```diff
@@ -85,7 +85,7 @@ generic-service:
     APPLICATIONINSIGHTS_CONFIGURATION_FILE: applicationinsights.json
 
     SPRING_DATASOURCE_HIKARI_MAXIMUM_POOL_SIZE: 15
-    SPRING_REPLICA_DATASOURCE_HIKARI_MAXIMUM_POOL_SIZE: 50
+    SPRING_REPLICA_DATASOURCE_HIKARI_MAXIMUM_POOL_SIZE: 25
```
Comment (petergphillips) [inline helm_deploy/prison-api/values.yaml:88]: hasn't gone about 8 in last 24 hours

Comment (steverendell) [reply helm_deploy/prison-api/values.yaml:88]: Could go lower even than this. My theory now is that the problematic pool entry was left idle for a very long time and this was why the setNetworkTimeout call failed. So the smaller the pool, the lower the chance of any pool entry being idle for an extended period. And that would mean  setting maxLifeTime would also work. Interestingly the hikari read.me https://github.com/brettwooldridge/HikariCP?tab=readme-ov-file#frequently-used says 'We strongly recommend setting this value'

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**